Design 1.0

kathleenconway opened the pull request:
Just a good old-fashioned setup assistant
- Outbound connection
- Contact mapping
- Debug log management
- Data sync scheduler
- Contact sync configuration

sean-appiphony commented on force-app/main/default/lwc/outboundConnectionStep/outboundConnectionStep.js:
```diff
@@ -0,0 +1,7 @@
+import { LightningElement } from 'lwc'
+
+export default class OutboundConnectionStep extends LightningElement {
+    loading = false
```
missing semicolon

kathleenconway replied:
✅

sean-appiphony commented on force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.html:
```diff
@@ -0,0 +1,166 @@
+                    </c-picklist>
+                </div>
+                <div class="slds-col slds-size_3-of-7">
+                    <!-- <template if:true={onceDaily}>
+                        <c-picklist>
+                        </c-picklist>
+                    </template>
+                    <template if:false={onceDaily}>
+                        <c-picklist
+                            disabled
+                            label="Job Start Time"
+                            options={nonDailyJobStartHourOptions}
+                            value="asap">
+                        </c-picklist>
+                    </template> -->
```
Will these be brought back at some point? If not, consider removing this comment block

kathleenconway replied:
🪓

sean-appiphony commented on force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.js:
```diff
@@ -0,0 +1,286 @@
+    @track objectLabel = 'Account'
+
+    updateObjectValue(event){
+        this.value = event.detail.value;
+        this.value = objectValue;
```
Won't this just set `this.value` attribute twice each time?

kathleenconway replied:
🤦‍♀️

sean-appiphony commented on force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.js:
```diff
@@ -0,0 +1,286 @@
+import saveData from '@salesforce/apex/setupAssistant.saveData';
+import getData from '@salesforce/apex/setupAssistant.getData';
+
+export default class FieldMappingStep extends LightningElement {
```
Generally speaking, consider grouping up similar elements within the document: attributes with attributes, getters/ setters with getters/ setters, methods with methods. Sometimes it makes sense to group certain blocks of demo code together, but this file appears a little random. Consider cleaning up for code clarity

kathleenconway replied:
I think this should be readable by humans other than myself now – let me know if anything needs to move around

sean-appiphony reviewed (commented):
A few things to look for throughout: 
1) Make sure all net new JS files have semicolons at the ends of statements
2) double check your class scoping prefixes to make sure they are consistent throughout the project

kathleenconway commented on force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.html:
```diff
@@ -0,0 +1,166 @@
+                    </c-picklist>
+                </div>
+                <div class="slds-col slds-size_3-of-7">
+                    <!-- <template if:true={onceDaily}>
+                        <c-picklist>
+                        </c-picklist>
+                    </template>
+                    <template if:false={onceDaily}>
+                        <c-picklist
+                            disabled
+                            label="Job Start Time"
+                            options={nonDailyJobStartHourOptions}
+                            value="asap">
+                        </c-picklist>
+                    </template> -->
```
Axed it

kathleenconway commented on force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.css:
```diff
@@ -0,0 +1,4 @@
+.bss-tooltip {
```
All but one of these needed to be cleaned out anyway so I changed it to a rl- prefix

kathleenconway commented on force-app/main/default/lwc/customPropertyEditor/customPropertyEditor.html:
```diff
@@ -0,0 +1,16 @@
+        label="Record ID"
+        value={recordID}>
+    </lightning-input>
+    <!-- <lightning-input
+        label="Object API Name"
+        value={objectAPI}>
+    </lightning-input> -->
```
🪓

kathleenconway commented on force-app/main/default/lwc/debugLogStep/debugLogStep.html:
```diff
@@ -0,0 +1,89 @@
+                <div class="slds-var-p-horizontal_large">
+                    <p class="slds-var-m-bottom_x-small">Your Rightsline Salesforce integration automatically creates activity logs for use in the debugging process.</p>
+                    <p>You can choose to log both success and error activity, or error activity alone.</p>
+                    <!-- <p>To improve app performance, you can set a limit on the number of debug log records that will be saved in this org. If the upper limit is hit, older debug logs will be deleted to make room for new ones.</p> -->
```
🪓

kathleenconway commented on force-app/main/default/lwc/editorConfigStep/editorConfigStep.js:
```diff
@@ -0,0 +1,7 @@
+import { LightningElement } from 'lwc'
+
+export default class OutboundConnectionStep extends LightningElement {
+    loading = false
```
✅

kathleenconway commented on force-app/main/default/lwc/debugLogStep/debugLogStep.js:
```diff
@@ -0,0 +1,23 @@
+
+export default class DebugLogStep extends LightningElement {
+
+    @track logLevel = 'DEBUG'
+    @track logCount = 0
+    @track logLimit = 10000
```
✅

kathleenconway commented on force-app/main/default/lwc/debugLogStep/debugLogStep.js:
```diff
@@ -0,0 +1,23 @@
+        return [
+            { label: 'All Activity', value: 'DEBUG' },
+            { label: 'Errors Only', value: 'ERROR' }
+        ]
```
✅

kathleenconway commented on force-app/main/default/lwc/customPropertyEditor/customPropertyEditor.js:
```diff
@@ -0,0 +1,15 @@
+            label: '1',
+            value: '1'
+        }
+    ]
```
✅

kathleenconway commented on force-app/main/default/lwc/editorConfigStep/editorConfigStep.js:
```diff
@@ -0,0 +1,7 @@
+import { LightningElement } from 'lwc'
+
+export default class OutboundConnectionStep extends LightningElement {
+    loading = false
```
🪓

kathleenconway commented on force-app/main/default/lwc/errorRetrySchedulerStep/errorRetrySchedulerStep.css:
```diff
@@ -0,0 +1,4 @@
+.bss-tooltip {
```
See above 🪓

kathleenconway commented on force-app/main/default/lwc/errorRetrySchedulerStep/errorRetrySchedulerStep.js:
```diff
@@ -0,0 +1,67 @@
+
+    openBatchModal() {
+        if(this.isActive) {
+            this.modalTitle = 'Deactivate Job';
+            this.modalButtonVariant = 'destructive'
+            this.modalButtonLabel = 'Deactivate'
```
✅ ✅ ✅

kathleenconway commented on force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.css:
```diff
@@ -0,0 +1,17 @@
+.rl-vertical_nav {
+    max-width: 180px;
+    min-width: 180px;
```
This was behaving peculiarly with both rem units and setting a single width, so I just put a slds-size class on the vertical nav instead – worked out better for my purposes anyway

kathleenconway commented on force-app/main/default/lwc/syncSchedulerStep/syncSchedulerStep.css:
```diff
@@ -0,0 +1,4 @@
+.bss-tooltip {
```
See above

kathleenconway commented on force-app/main/default/lwc/syncSchedulerStep/syncSchedulerStep.js:
```diff
@@ -0,0 +1,79 @@
+        }
+    ]
+
+    updateValue(event){
+        let value = event.detail.value;
+        let label = event.detail.label;
+
+        this.value = value;
+        this.label = label;
+    }
```
✅

kathleenconway commented on force-app/main/default/lwc/packageLogConfigStep/packageLogConfigStep.js:
```diff
@@ -0,0 +1,40 @@
+            label:'All Activity',
+            value:'allActivity'
+        }
+    ]
```
✅

kathleenconway commented:
- Cleaned out commented-out code & superfluous css classes
- Added semicolons to ends of statements
- Got rid of static sizing on vertical nav of mapping step
- Major rehab of fieldMappingStep.js
